vault/resource_database_secret_backend_connection_test.go

```diff
@@ -538,31 +549,29 @@ func TestAccDatabaseSecretBackendConnectionTemplatedUpdateExcludePassword_mysql(
 	createMySQSUser(t, db, testUsername, testPassword)
 	defer deleteMySQLUser(t, db, testUsername)
 
-	client := testProvider.Meta().(*api.Client)
-
 	resource.Test(t, resource.TestCase{
 		Providers:    testProviders,
 		PreCheck:     func() { testutil.TestAccPreCheck(t) },
 		CheckDestroy: testAccDatabaseSecretBackendConnectionCheckDestroy,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccDatabaseSecretBackendConnectionConfigTemplated_mysql(name, backend, testConnURL, testUsername, testPassword, 0),
+				Config: testAccDatabaseSecretBackendConnectionConfigTemplated_mysql(name, backend, testConnURL, username, password, 0),
 				Check: testComposeCheckFuncCommonDatabaseSecretBackend(name, backend, pluginName,
 					resource.TestCheckResourceAttr(testDefaultDatabaseSecretBackendResource, "allowed_roles.#", "2"),
 					resource.TestCheckResourceAttr(testDefaultDatabaseSecretBackendResource, "allowed_roles.0", "dev"),
 					resource.TestCheckResourceAttr(testDefaultDatabaseSecretBackendResource, "allowed_roles.1", "prod"),
 					resource.TestCheckResourceAttr(testDefaultDatabaseSecretBackendResource, "verify_connection", "true"),
 					resource.TestCheckResourceAttr(testDefaultDatabaseSecretBackendResource, "mysql.0.connection_url", testConnURL),
+					resource.TestCheckResourceAttr(testDefaultDatabaseSecretBackendResource, "mysql.0.username", username),
+					resource.TestCheckResourceAttr(testDefaultDatabaseSecretBackendResource, "mysql.0.password", password),
 					resource.TestCheckResourceAttr(testDefaultDatabaseSecretBackendResource, "mysql.0.max_connection_lifetime", "0"),
-					resource.TestCheckResourceAttr(testDefaultDatabaseSecretBackendResource, "data.%", "2"),
-					resource.TestCheckResourceAttr(testDefaultDatabaseSecretBackendResource, "data.username", testUsername),
-					resource.TestCheckResourceAttr(testDefaultDatabaseSecretBackendResource, "data.password", testPassword),
 				),
 			},
 			{
 				Config: testAccDatabaseSecretBackendConnectionConfigTemplated_mysql(name, backend, testConnURL, testUsername, testPassword, 10),
 				PreConfig: func() {
 					path := fmt.Sprintf("%s/rotate-root/%s", backend, name)
```

Comment:
@benashz for context, this is the test I'm confused about: this test connects to DB using `admin` or `vault superuser` creds, creates a new user `testUsername` in the DB w/ a new password, and establishes a DB connection in Vault with the new user. In step 2, it first rotates the credentials of the connection (changes the password for `testUsername`), and then confirms the connection is still established successfully.

I don't understand how that works... If the credentials have been updated, how does the same password work in establishing a connection? The test on my branch fails on step 2 after rotating credentials, and though this seems expected to me, I don't understand how it passes on `main`

Reply:
The difference is that in my case the `username` and `password` are passed in the `mysql` block as opposed to the `data` block

Reply:
Issue seems to have been caused by a test calling `rotate-root` on the `root` MySQL user, rather than the intermediary admin user. Fixed in #1340